crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -320,7 +322,32 @@ impl EventCache {
 }
 
 type AllEventsMap = BTreeMap<OwnedEventId, (OwnedRoomId, SyncTimelineEvent)>;
-type RelationsMap = BTreeMap<OwnedEventId, BTreeSet<OwnedEventId>>;
+type RelationsMap = BTreeMap<OwnedEventId, BTreeSet<RelatedEvent>>;
+
+/// Contains relationship information for a related event.
+#[derive(Clone, Eq, PartialEq)]
+struct RelatedEvent {
+    related_event_id: OwnedEventId,
+    relation_type: RelationType,
+}
+
+impl RelatedEvent {
+    fn new(related_event_id: OwnedEventId, relation_type: RelationType) -> Self {
+        RelatedEvent { related_event_id, relation_type }
+    }
+}
+
+impl PartialOrd<Self> for RelatedEvent {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
```

Comment:
Can't you `#[derive(PartialOrd)` on `RelatedEvent`?

Reply:
I don't think so, `RelationType` comes from Ruma and doesn't implement it, so I need to manually exclude it.